clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/helpers/AbstractServerGroupNameResolver.groovy

```diff
@@ -44,6 +47,17 @@ abstract class AbstractServerGroupNameResolver extends NameBuilder {
     TaskRepository.threadLocalTask.get()
   }
 
+  String resolveLatestServerGroupName(String clusterName) {
+    List<TakenSlot> takenSlots = getTakenSlots(clusterName)
+
+    // Attempt to find the server group created most recently.
+    def latestServerGroup = takenSlots?.max { a, b ->
+      a.createdTime <=> b.createdTime
+    }
+
+    return latestServerGroup
+  }
+
   String resolveNextServerGroupName(String application, String stack, String details, Boolean ignoreSequence) {
     Integer nextSequence = 0
     String clusterName = combineAppStackDetail(application, stack, details)
```

Comment:
Looks like the next 6 lines can be replaced with a call to `resolveLatestServerGroupName()`.